src/transport/HighPressureGasTransport.cpp

```diff
@@ -109,216 +180,141 @@ double HighPressureGasTransport::thermalConductivity()
+            d[ld*j + i] = P_corr_ij*(rp * m_bdiff(i,j));
         }
     }
 }
 
-void HighPressureGasTransport::getMultiDiffCoeffs(const size_t ld, double* const d)
+double HighPressureGasTransport::viscosity()
 {
-    // Not currently implemented.  m_Lmatrix inversion returns NaN.  Needs to be
-    //   fixed.  --SCD - 2-28-2014
-    throw NotImplementedError("HighPressureGasTransport:getMultiDiffCoeffs");
-    // Calculate the multi-component Stefan-Maxwell diffusion coefficients,
-    // based on the Takahashi-correlation-corrected binary diffusion coefficients.
-
-    // update the mole fractions
-    update_C();
-
-    // update the binary diffusion coefficients
-    update_T();
-    updateThermal_T();
-
-    // Correct the binary diffusion coefficients for high-pressure effects; this
-    // is basically the same routine used in 'getBinaryDiffCoeffs,' above:
-    size_t nsp = m_thermo->nSpecies();
-    vector<double> molefracs(nsp);
-    m_thermo->getMoleFractions(&molefracs[0]);
-    update_T();
-    // Evaluate the binary diffusion coefficients from the polynomial fits -
-    // this should perhaps be preceded by a check for changes in T, P, or C.
-    updateDiff_T();
-
-    if (ld < m_nsp) {
-        throw CanteraError("HighPressureGasTransport::getMultiDiffCoeffs",
-                           "ld is too small");
-    }
-    for (size_t i = 0; i < m_nsp; i++) {
-        for (size_t j = 0; j < m_nsp; j++) {
-            // Add an offset to avoid a condition where x_i and x_j both equal
-            //   zero (this would lead to Pr_ij = Inf):
-            double x_i = std::max(Tiny, molefracs[i]);
-            double x_j = std::max(Tiny, molefracs[j]);
-            x_i = x_i/(x_i+x_j);
-            x_j = x_j/(x_i+x_j);
-            double Tr_ij = m_temp/(x_i*Tcrit_i(i) + x_j*Tcrit_i(j));
-            double Pr_ij = m_thermo->pressure()/(x_i*Pcrit_i(i) + x_j*Pcrit_i(j));
-
-            double P_corr_ij;
-            if (Pr_ij < 0.1) {
-                P_corr_ij = 1;
-            }else {
-                P_corr_ij = setPcorr(Pr_ij, Tr_ij);
-                if (P_corr_ij<0) {
-                    P_corr_ij = Tiny;
-                }
-            }
-
-            m_bdiff(i,j) *= P_corr_ij;
-        }
-    }
-    m_bindiff_ok = false; // m_bdiff is overwritten by the above routine.
+    LucasMixtureParameters params;
+    compute_mixture_parameters(params);
+
+    // This is η*ξ
+    double nondimensional_viscosity = high_pressure_nondimensional_viscosity(params.Tr_mix,
+                                                                             params.Pr_mix,
+                                                                             params.FP_mix_o,
+                                                                             params.FQ_mix_o,
+                                                                             params.P_vap_mix,
+                                                                             params.Pc_mix);
```

Comment:
When the following line indent is this extreme, I'd suggest the following line wrap style instead:
```suggestion
    double nondimensional_viscosity = high_pressure_nondimensional_viscosity(
        params.Tr_mix, params.Pr_mix, params.FP_mix_o, params.FQ_mix_o,
        params.P_vap_mix, params.Pc_mix);
```